src/modules/LeaderboardApi.js

```diff
@@ -2,13 +2,22 @@ import ExtendClassApi from './extendClassApi.js';
 
 class LeaderboardApi {
   init = async () => {
+    this.displayLoadingScreen();
     this.extendClassApi = new ExtendClassApi();
     this.gameId = await this.extendClassApi.getGameId();
     if (!this.gameId) {
-      this.addGame('Micro Game');
+      await this.addGame('Micro Game');
     }
     this.baseUri = 'https://us-central1-js-capstone-backend.cloudfunctions.net/api/games/';
-    this.getGameData();
+    await this.getGameData();
+  }
+
+  displayLoadingScreen = () => {
+    document.querySelector('#loading-bar').style.display = 'block';
+  }
+
+  hideLoadingScreen = () => {
+    document.querySelector('#loading-bar').style.display = 'none';
   }
```

Comment:
- [ ] [OPTIONAL] How about instead of using two functions definitions ?  we can only use one function 🤔 

```suggestion
  loadingScreen = (status) => {
    document.querySelector('#loading-bar').style.display = status;
  }
```

```js
// when display
loadingScreen('block')
```

```js
// when hide
loadingScreen('none')
```